From 0f76d88556a56d3676974c7a1c09736be52dadde Mon Sep 17 00:00:00 2001
From: Hugo Manrique <hugmanrique@gmail.com>
Date: Mon, 16 Jul 2018 12:42:20 +0200
Subject: [PATCH] Avoid item merge if stack size above max stack size


diff --git a/src/main/java/net/minecraft/server/EntityItem.java b/src/main/java/net/minecraft/server/EntityItem.java
index 0b7fc327f..099bc7551 100644
--- a/src/main/java/net/minecraft/server/EntityItem.java
+++ b/src/main/java/net/minecraft/server/EntityItem.java
@@ -177,6 +177,11 @@ public class EntityItem extends Entity implements HopperPusher {
     // Spigot end
 
     private void x() {
+        // Paper start - avoid item merge if stack size above max stack size
+        ItemStack stack = getItemStack();
+        if (stack.getCount() >= stack.getMaxStackSize()) return;
+        // Paper end
+
         // Spigot start
         double radius = world.spigotConfig.itemMerge;
         Iterator iterator = this.world.a(EntityItem.class, this.getBoundingBox().grow(radius, radius, radius)).iterator();
-- 
2.35.1

